unit-tests/py/common.py

```diff
@@ -0,0 +1,117 @@
+import os, sys, subprocess, traceback
+import pyrealsense2 as rs
+
+n_assertions = 0
+n_failed_assertions = 0
+n_tests = 0
+n_failed_tests = 0
+test_failed = False
+
+# If this is the first time running this script we set the wanted enviroment, However it is impossible to change the current running 
+# environment so we rerun the script in a child process that inherits the enviroment we set
+def set_env_vars(env_vars):
+    if len(sys.argv) < 2:
+        for env_var, val in env_vars.items():
+            os.environ[env_var] = val
+        sys.argv.append("rerun")
+        cmd = ["py", "-3"] + sys.argv
+        p = subprocess.run( cmd, stderr=subprocess.PIPE, universal_newlines=True )
+        exit(p.returncode)
+
+# Returns the first device that was found, if no device is found the test is aborted
+def get_first_device():
+    c = rs.context()
+    if not c.devices.size():  # if no device is connected we don't run the AC test
+        print("No device found, aborting AC test")
+        exit(0)
+    return c.devices[0]
+
+# Functions for asserting test cases:
+# Receives an expression which is an assertion. If false the assertion failed
+def require(exp):  
+    global n_assertions, n_failed_assertions, test_failed
+    n_assertions += 1
+    if not exp:
+        n_failed_assertions += 1
+        test_failed = True
+        print("Require failed")
+        for line in traceback.format_stack():
+            print(line.strip())
+
+# Receives the resulted value and the expected value and asserts they are equal
+def require_equal(result, expected):
+    global n_assertions, n_failed_assertions, test_failed
+    n_assertions += 1
+    if not result == expected:
+        n_failed_assertions += 1
+        test_failed = True
+        print("Result was: " + result + "\nBut we expected: " + expected)
+        for line in traceback.format_stack():
+            print(line.strip())
+
+# This function should never be reached, it receives the number of assertion that were 
+# skipped if it was reached
+def require_no_reach(skipped):
+    require(False)
+
+# Receives 2 lists and asserts they are identical
+def require_equal_lists(result, expected):
+    global n_assertions, n_failed_assertions, test_failed
+    n_assertions += 1
+    failed = False
+    if not len(result) == len(expected):
+        n_failed_assertions += 1
+        test_failed = True
+        failed = True
+        print("Require equal lists failed due to lists of different sizes:")
+        print("The resulted list has " + str(len(result)) + " elements, but the expected list has " + str(len(expected)) + " elements")
+    i = 0
+    for res, exp in zip(result, expected):
+        n_assertions += 1
+        if not res == exp:
+            n_failed_assertions += 1
+            test_failed = True
```

Comment:
Move the common lines into the block in line 78

Reply:
I don't understand. This block is for every different element so that if there's more than 1 I can see all of them. The block at the end is if there was any mistake so it only happens once. I can't combine them

Reply:
`test_failed` can be set to True only when `failed` is checked.
Also, I'm not sure that each element that differs counts as an assertion -- there was one `require` issues, so one assertion...

Reply:
Ok, I'll change this. Was considering it anyway